go_modules/lib/dependabot/go_modules/update_checker/latest_version_finder.rb

```diff
@@ -73,23 +75,22 @@ def fetch_lowest_security_fix_version
         def available_versions
           SharedHelpers.in_a_temporary_directory do
             SharedHelpers.with_git_configured(credentials: credentials) do
-              File.write("go.mod", go_mod.content)
+              manifest = parse_manifest
+
+              # Set up an empty go.mod so 'go list -m' won't attempt to download dependencies. This
```

Comment:
One other observation from testing is that we can also run `go list -m -versions` without a `go.mod` and it works the same way. However, it won't know about the exclude directives and won't be able to apply them for us so creating a `go.mod` with the excludes is still preferred.